web/src/app/selection/MaterialSelect.tsx

```diff
@@ -151,7 +151,7 @@ export default function MaterialSelect(
           setInputValue('')
         }
       }}
-      onBlur={getLabel}
+      onBlur={() => setInputValue(multiple || !value ? '' : getLabel())}
```

Comment:
`getLabel` should handle returning either an empty string or the current value's label based on if it's set to multiple or not:

```
const getLabel = (): string => Array.isArray(value) ? value[0]?.label ?? '' : value?.label ?? ''
```

```suggestion
      onBlur={() => setInputValue(getLabel())}
```

Reply:
Ah, then the original `onBlur` handler may have been correct and `getLabel` is broken instead.

I'll adjust and add a comment to `getLabel`

Reply:
Gotcha, sounds good — That probably sounds like the case, I had originally made the `getLabel` function as a simple helper since I was using that logic in more than 1 place with the previous changes that likely caused the regression

Reply:
I think you would only need to call `setInputValue` as the net new change, as the bug looks like it was because we weren't actually doing anything onBlur except saying "hey this is the new string lol"

Reply:
It might be a little deeper, because there's also when to call `onInputChange` which updates the query to the server which isn't working properly either. But calling it all the time also poses an issue because we want the search string to be empty if the input == the selected value.

With what I have pushed up now, the only issue is you get "No options" on a single-select with a selected value. I *think* that's the only outstanding bit.

Going to take a break from it for a bit and then look at it again.